### PR TITLE
updating Pill state and removing event

### PR DESCRIPTION
**Pull request by antontim4enko**

### CODE REVIEW

File: src/screens/Pills/scenes/AddPill/AddPill.tsx [hunk 1]
```diff
@@ -177,7 +178,7 @@ export class AddPillComponent extends React.Component<
   private onDeleteTime = (index: number) => {
     const timeOptions = [...this.state.pill.timeOptions];
     timeOptions.splice(index, 1);
-    this.setState((state) => ({
+    this.setState((state: {pill: IPill}) => ({
```
Comment (rumbaa01) [inline src/screens/Pills/scenes/AddPill/AddPill.tsx]: We have another type fo state -> IAddPillState

File: src/redux/reducers/eventReducer.ts [hunk 2]
```diff
@@ -16,24 +16,31 @@ interface IAddAction {
 }
 
 interface IDeleteAction {
-  payload: {
-    index: number;
-  };
+  payload: string;
+}
+interface IGetAction {
```
Comment (rumbaa01) [inline src/redux/reducers/eventReducer.ts]: Let's do generic for actions

File: src/redux/reducers/pillsReducer.ts [hunk 3]
```diff
@@ -27,26 +26,23 @@ interface IActionTakePill {
 }
 
 const handlers = {
-  [ADD_PILL]: (state: IState, action: IActionAddPill) => {
+  [GET_PILLS_ASYNC]: (state: IState, action: any) => {
```
Comment (rumbaa01) [inline src/redux/reducers/pillsReducer.ts]: why any?

File: src/sagas/addEvent.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+interface IAddEventPayload {
+    event: IEvent,
+    id: string
+}
+
+function* addEventAsync(payload: IAddEventPayload) {
+  try {
+    yield firebaseService.setToBase(payload.event, payload.id, '/events');
+    debugger;
```
Comment (dimagreat) [inline src/sagas/addEvent.ts]: remove

File: src/sagas/addEvent.ts [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+import { firebaseService } from '../services/firebaseService';
+import { ADD_EVENT, addEventAsyncAction } from '../redux/actions/eventAction';
+import { IEvent } from 'screens/Events/scenes/EditEvent';
+
+interface IAddEventPayload {
+    event: IEvent,
+    id: string
+}
+
+function* addEventAsync(payload: IAddEventPayload) {
```
Comment (dimagreat) [inline src/sagas/addEvent.ts]: Move saga to according scene

File: tsconfig.json [hunk 6]
```diff
@@ -38,14 +38,14 @@
...
     "baseUrl": "./src" /* Base directory to resolve non-absolute module names. */,
-    "paths": {
-      "components/*": ["./components/*"],
-      "services/*": ["./services/*"],
-      "constants/*": ["./constants/*"],
-      "screens/*": ["./screens/*"],
-      "redux/*": ["./redux/*"],
-      "assets/*": ["./assets/*"]
-    }, // "rootDirs": [],                        /* List of root folders whose combined content represents the structure of the project at runtime. */ // "typeRoots": [],                       /* List of folders to include type definitions from. */ // "types": [],                           /* Type declaration files to be included in compilation. */ // "allowSyntheticDefaultImports": true,  /* Allow default imports from modules with no default export. This does not affect code emit, just typechecking. */
+    // "paths": {
```
Comment (dimagreat) [inline tsconfig.json]: Remove

File: src/screens/Events/scenes/Events/Events.tsx [hunk 7]
```diff
@@ -10,42 +10,34 @@ import { strings } from '../../../../constants/strings';
...
+  componentDidMount() {
+    this.props.getEvents();
   }
-  setEventToState = (event: any) => {
-    this.setState({
-      events: [...event],
-    });
-  };
+
   renderPill = (item: any) => {
```
Comment (rumbaa01) [inline src/screens/Events/scenes/Events/Events.tsx]: any?

File: src/screens/Pills/scenes/Pills/Pills.tsx [hunk 8]
```diff
@@ -31,12 +31,12 @@ export class PillsComponent extends React.PureComponent<IPillsComponentProps> {
...
+  componentDidMount() {
+    this.props.getPillsDispatch();
   }
   setPillToState = (pills: IPill[]) => {
     this.setState({
-      pills: [...this.state.pills, pills],
+      pills: [...pills],
     });
   };
   renderPill = (item: any) => {
```
Comment (rumbaa01) [inline src/screens/Pills/scenes/Pills/Pills.tsx]: any?

File: src/redux/actions/eventAction.ts [hunk 9]
```diff
@@ -1,19 +1,43 @@
...
+export const GET_EVENTS_ASYNC = 'GET_EVENTS_ASYNC';
+export const DELETE_EVENT_ASYNC = 'DELETE_EVENT_ASYNC';
 
 import { IEvent } from '../../screens/Events/scenes/EditEvent/EditEvent';
 
-export const addEvent = (event: IEvent) => {
+export const addEventAsyncAction = (event: IEvent) => {
   return {
-    type: ADD_EVENT,
+    type: ADD_EVENT_ASYNC,
```
Comment (dimagreat) [inline src/redux/actions/eventAction.ts]: I didn't get what exactly event we using `ADD_EVENT_ASYNC` or `ADD_EVENT` in reducer we have only async event. Please clean code from unused functions and variables

**Pull request merged**